proto/redpanda/api/dataplane/v1/quota.proto

```diff
@@ -0,0 +1,222 @@
+    repeated Quota.Value values = 2;
+  }
+  repeated QuotaEntry quotas = 1;
+}
+
+message RequestEntity {
+  option (buf.validate.message).cel = {
+    id: "name_required_for_entity_request_type_name"
+    message: "entity_name is required when entity_request_type is ENTITY_REQUEST_TYPE_NAME"
+    expression: "this.entity_request_type != 1 || this.entity_name != ''"
+  };
+  option (buf.validate.message).cel = {
+    id: "client_id_prefix_requires_name_request_type"
+    message: "CLIENT_ID_PREFIX entity_type requires ENTITY_REQUEST_TYPE_NAME, not ENTITY_REQUEST_TYPE_DEFAULT"
+    expression: "this.entity_type != 3 || this.entity_request_type == 1"
+  };
+
+  Quota.EntityType entity_type = 1 [
+    (buf.validate.field).enum = {
+      defined_only: true
+      not_in: [
+        0,
+        1,
+        4,
+        5
+      ] /* Disallow unspecified, any, user, and ip */
+    },
+    (buf.validate.field).required = true
+  ];
+
+  // Entity can be defined by its name or use the default value.
+  enum EntityRequestType {
+    ENTITY_REQUEST_TYPE_UNSPECIFIED = 0;
+    ENTITY_REQUEST_TYPE_NAME = 1;
+    ENTITY_REQUEST_TYPE_DEFAULT = 2;
+  }
+
+  EntityRequestType entity_request_type = 2 [(buf.validate.field).required = true];
+
+  // Either the client ID or the client ID prefix, depending on the entity_request_type
+  // If entity_request_type is ENTITY_REQUEST_TYPE_DEFAULT, this field can be empty
+  string entity_name = 3;
+}
+
+message CreateQuotaRequest {
+  message RequestValue {
+    Quota.ValueType value_type = 1 [
+      (buf.validate.field).enum = {
+        defined_only: true
+        not_in: [
+          0,
+          1
+        ] /* Disallow unspecified and any */
+      },
+      (buf.validate.field).required = true
+    ];
+    int64 value = 2;
+  }
+
+  // To specify multiple entities to apply the same quota
+  repeated RequestEntity entities = 1 [(buf.validate.field).required = true];
+  repeated RequestValue values = 2 [(buf.validate.field).required = true];
+}
+
+message CreateQuotaResponse {}
+
+message DeleteQuotaRequest {
+  option (buf.validate.message).cel = {
+    id: "value_type_not_unspecified_or_any"
+    message: "value_type cannot be VALUE_TYPE_UNSPECIFIED (0) or VALUE_TYPE_ANY (1)"
+    expression: "this.value_type != 0 && this.value_type != 1"
+  };
+
+  RequestEntity entity = 1 [(buf.validate.field).required = true];
+
+  // List of values that should be deleted for the specified entities
+  Quota.ValueType value_type = 2 [(buf.validate.field).required = true];
+}
+
+message DeleteQuotaResponse {}
+
+service QuotaService {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_tag) = {
+    name: "Quotas"
+    description: "Manage client throughput limits. See [Redpanda Client Throughput Limits](https://docs.redpanda.com/current/manage/cluster-maintenance/manage-throughput/#client-throughput-limits) for more information."
+  };
+
+  rpc ListQuotas(ListQuotasRequest) returns (ListQuotasResponse) {
```

Comment:
We probably want to add pagination here, see other requests. The Kafka API doesn't support pagination but we offer it for resource-constrained frontend clients, including the possibility to run filters on the backend in case there or hundreds or thousands of resources.